### PR TITLE
Support partition filters for cdf queries in v0.5

### PR DESCRIPTION
**Pull request by linzhou-db**
Support partition filters for cdf queries

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala [hunk 1]
```diff
@@ -164,20 +166,56 @@ private[sharing] case class RemoteDeltaCDFAddFileIndex(
...
+  override def listFiles(
+    partitionFilters: Seq[Expression],
+    dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    val updatedFiles = deltaTableFiles.addFilesForCdf.map { a =>
+      AddFileForCDF(a.url, a.id, a.getPartitionValuesInDF, a.size, a.version, a.timestamp, a.stats)
+    }
+    val columnFilter = getColumnFilter(partitionFilters)
+    val implicits = params.spark.implicits
+    import implicits._
+    makePartitionDirectories(updatedFiles.toDS().filter(columnFilter).as[AddFileForCDF].collect())
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:179]: same question as in the other PR.

Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:184]: added comments

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala [hunk 2]
```diff
@@ -166,20 +168,65 @@ private[sharing] case class RemoteDeltaCDFAddFileIndex(
...
-      CDFColumnInfo.getInternalPartitonSchemaForCDFAddRemoveFile) {}
+      CDFColumnInfo.getInternalPartitonSchemaForCDFAddRemoveFile) {
+  override def listFiles(
+    partitionFilters: Seq[Expression],
+    dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    // Need to apply getPartitionValuesInDF to each file, to be consistent with
+    // makePartitionDirectories and partitionSchema. So that partitionFilters can be correctly
+    // applied.
+    val updatedFiles = deltaTableFiles.addFilesForCdf.map { a =>
+      AddFileForCDF(a.url, a.id, a.getPartitionValuesInDF, a.size, a.version, a.timestamp, a.stats)
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:179]: As discussed, perhaps add a comment in getPartitionValuesInDF that the function is idempotent, and calling it multiple times does not change its output.

**Review by chakankardb — APPROVED** [review]

File: spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala (hunk 2 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala:181]: added

**Pull request merged**